### PR TITLE
BZ1702011:[Docs][Admin] engine support for SNMPv3 traps

### PR DESCRIPTION
**Pull request by stoobie**
RFE

Fixes BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1702011

Changes proposed in this pull request:

- Some minor modular documentation edits

- Adding info necessary for SNMPv3 traps

Preview of topics:
https://ovirt.github.io/ovirt-site/previews/2674/documentation/administration_guide/index.html#Parameters_for_event_notifications_in_notifier
https://ovirt.github.io/ovirt-site/previews/2674/documentation/administration_guide/index.html#Configuring_the_Red_Hat_Enterprise_Virtualization_Manager_to_Send_SNMP_Traps
https://ovirt.github.io/ovirt-site/previews/2674/documentation/administration_guide/index.html#SNMPv3-traps-configuration-settings

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @stoobie 

This pull request needs review by: @eli

### CODE REVIEW

**Comment by stoobie** [comment]
@janosdebugs Any idea on how to enable callout formatting so that this sample file appears like the sample files in OCP docs?

https://ovirt.github.io/ovirt-site/previews/2674/documentation/administration_guide/index.html#SNMPv3-traps-configuration-settings

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+{virt-product-fullname} supports SNMP version 2 and version 3. SNMP version 3 supports the following security levels:
 
-* The IP addresses or fully qualified domain names of machines that will act as SNMP managers. Optionally, determine the port through which {engine-name} receives trap notifications; by default, this is UDP port 162.
+*NoAuthNoPriv*:: SNMP traps are sent without any authorization or privacy.
+*AuthNoPriv*:: SNMP traps are sent with password authorization but no privacy.
+*AuthPriv*:: SNMP traps are sent with password authorization and privacy.
+
+.Prerequisites
+
+* One or more external SNMP managers is configured to receive traps.
```
Comment (emarcusRH) [inline source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:14]: One or more external SNMP managers **are** configured to receive traps

Comment (emarcusRH) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:14]: other that this grammatical suggestion, 
/lgtm

**Review by emesika — COMMENTED** [review]
lgtm

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 1 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc]: Fixed.

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc [hunk 2]
```diff
@@ -1,46 +1,107 @@
...
+*AuthNoPriv*:: SNMP traps are sent with password authorization but no privacy.
+*AuthPriv*:: SNMP traps are sent with password authorization and privacy.
+
+.Prerequisites
+
+* One or more external SNMP managers are configured to receive traps.
+
+* The IP addresses or fully qualified domain names of machines that will act as SNMP managers. Optionally, determine the port through which {engine-name} receives trap notifications. The default is UDP port 162.
 
 * The SNMP community. Multiple SNMP managers can belong to a single community. Management systems and agents can communicate only if they are within the same community. The default community is `public`.
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:18]: relevant only on SNMP v1/v2c
SNMPv3 calls communities "Groups"

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc [hunk 3]
```diff
@@ -1,46 +1,107 @@
...
 # vi /etc/ovirt-engine/notifier/notifier.conf.d/20-snmp.conf
 ----
 +
-. Specify the SNMP manager(s), the SNMP community, and the OID in the following format:
+[TIP]
+====
+Copy the default SNMP settings from the events notification daemon configuration file `/usr/share/ovirt-engine/services/ovirt-engine-notifier/ovirt-engine-notifier.conf`. This file includes inline comments for all settings.
+====
+
+. Specify the SNMP manager(s), the SNMP community, and the OID in the format in this example:
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:46]: only relevant on v1/v2c

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc [hunk 4]
```diff
@@ -1,46 +1,107 @@
...
+
+* One or more external SNMP managers are configured to receive traps.
+
+* The IP addresses or fully qualified domain names of machines that will act as SNMP managers. Optionally, determine the port through which {engine-name} receives trap notifications. The default is UDP port 162.
 
 * The SNMP community. Multiple SNMP managers can belong to a single community. Management systems and agents can communicate only if they are within the same community. The default community is `public`.
 
 * The trap object identifier for alerts. The {virt-product-fullname} {engine-name} provides a default OID of 1.3.6.1.4.1.2312.13.1.1. All trap types are sent, appended with event information, to the SNMP manager when this OID is defined. Note that changing the default trap prevents generated traps from complying with the {engine-name}'s management information base.
 
+* An SNMP username and password.
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:22]: I would complement here with the other security scenarios:
An SNMP username and passphrase if security level chosen is 1.
An SNMP username and passphrase if security level chosen is 2.
An SNMP username, password and private passphrase if security level chosen is 3.

Smth like that ^

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc [hunk 5]
```diff
@@ -1,46 +1,107 @@
...
+Copy the default SNMP settings from the events notification daemon configuration file `/usr/share/ovirt-engine/services/ovirt-engine-notifier/ovirt-engine-notifier.conf`. This file includes inline comments for all settings.
+====
+
+. Specify the SNMP manager(s), the SNMP community, and the OID in the format in this example:
 +
 [options="nowrap" subs="normal"]
 ----
-SNMP_MANAGERS="_manager1.example.com_ _manager2.example.com:162"_
+SNMP_MANAGERS="_manager1.example.com_ _manager2.example.com:162_"
 SNMP_COMMUNITY=public
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:51]: Not relevant on v3

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 2 above)
Comment (santos1709) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:18]: I think it just worth mentioning somewhere

File: source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc [hunk 6]
```diff
@@ -34,9 +34,24 @@ The event notifier configuration file can be found in */usr/share/ovirt-engine/s
 |MAIL_REPLY_TO |none |This variable specifies reply-to addresses in RFC822 format on sent mail, if supported by the mail server.
 |MAIL_SEND_INTERVAL |1 |The number of SMTP messages to be sent for each IDLE_INTERVAL
 |MAIL_RETRIES |4 |The number of times to attempt to send an email before failing.
-|SNMP_MANAGER |none |The IP addresses or fully qualified domain names of machines that will act as the SNMP managers. Entries must be separated by a space and can contain a port number. For example, `manager1.example.com manager2.example.com:164`
-|SNMP_COMMUNITY |public |The default SNMP community.
+|SNMP_MANAGERS |none |The IP addresses or fully qualified domain names of machines that will act as the SNMP managers. Entries must be separated by a space and can contain a port number. For example, `manager1.example.com manager2.example.com:164`
+|SNMP_COMMUNITY |public |The SNMP community.
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc:38]: only relevant on v1/v2c

File: source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc [hunk 7]
```diff
@@ -34,9 +34,24 @@ The event notifier configuration file can be found in */usr/share/ovirt-engine/s
...
 |SNMP_OID |1.3.6.1.4.1.2312.13.1.1 |The default trap object identifiers for alerts. All trap types are sent, appended with event information, to the SNMP manager when this OID is defined. Note that changing the default trap prevents generated traps from complying with the {engine-name}'s management information base.
+|SNMP_VERSION |2 |Defines which version of SNMP to use. SNMP version 2 and version 3 traps are supported. Possible values: `2` or `3`.
+|SNMP_ENGINE_ID |none |(SNMPv3) The {engine-name} ID used for SNMPv3 traps. ???How do I get this ID???
```
Comment (santos1709) [inline source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc:41]: This ID comes from the switch/device to be connected through SNMP

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 3 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:46]: @santos1709 I think you're comment here only refers to the community, correct?

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 4 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:22]: @santos1709 Take a look now.

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 5 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:54]: @santos1709 What's not relevant on v3? the entire SNMP_MANAGERS line?

File: source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc (hunk 6 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc:38]: Fixed.

File: source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc (hunk 7 above)
Comment (stoobie) [reply source/documentation/administration_guide/topics/Parameters_for_event_notifications_in_ovirt-engine-notifier.adoc:41]: Fixed.

**Review by emarcusRH — APPROVED** [review]
/lgtm

File: source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc (hunk 5 above)
Comment (santos1709) [reply source/documentation/administration_guide/topics/Configuring_the_Red_Hat_Virtualization_Manager_to_Send_SNMP_Traps.adoc:54]: the community, selected wrong line

**Pull request merged**